Sources/TuistGenerator/Generator/LinkGenerator.swift

```diff
@@ -136,22 +134,30 @@ final class LinkGenerator: LinkGenerating {
                                    sourceRootPath: sourceRootPath)
     }
 
-    func generatePackages(target _: Target,
+    func generatePackages(target: Target,
                           pbxTarget: PBXTarget,
-                          pbxProject: PBXProject,
-                          packages: [PackageNode]) throws {
-        try packages.forEach { package in
-            switch package.packageType {
-            case let .local(path: packagePath, productName: productName):
-                _ = try pbxProject.addLocalSwiftPackage(path: Path(packagePath.pathString),
-                                                        productName: productName,
-                                                        targetName: pbxTarget.name,
-                                                        addFileReference: false)
-            case let .remote(url: url, productName: productName, versionRequirement: versionRequirement):
-                _ = try pbxProject.addSwiftPackage(repositoryURL: url,
-                                                   productName: productName,
-                                                   versionRequirement: versionRequirement.xcodeprojValue,
-                                                   targetName: pbxTarget.name)
+                          pbxproj: PBXProj) throws {
+        for dependency in target.dependencies {
+            switch dependency {
```

Comment:
I wonder if it was beneficial to add this to `XcodeProj`, something along the lines of `addSwiftPackageProduct`.

Reply:
I did consider this but was unsure since the API is quite high level. I will write an extension function and let the API mature more before moving it into XcodeProj:

```swift
extension PBXTarget {
    
    func addSwiftPackageProduct(productName: String, pbxproj: PBXProj) throws {
        
        let productDependency = XCSwiftPackageProductDependency(productName: productName)
        pbxproj.add(object: productDependency)
        packageProductDependencies.append(productDependency)
        
        // Build file
        let buildFile = PBXBuildFile(product: productDependency)
        pbxproj.add(object: buildFile)
        
        // Link the product
        guard let frameworksBuildPhase = try frameworksBuildPhase() else {
            throw "No frameworks build phase"
        }
        
        frameworksBuildPhase.files?.append(buildFile)
        
    }
    
}
```